builder/lxc/step_lxc_create.go

```diff
@@ -17,51 +19,83 @@ type stepLxcCreate struct{}
 func (s *stepLxcCreate) Run(state multistep.StateBag) multistep.StepAction {
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
-
-	name := config.ContainerName
+	errorHandler := func(err error) {
+		state.Put("error", err)
+		ui.Error(err.Error())
+	}
 
 	// TODO: read from env
 	lxc_dir := "/var/lib/lxc"
+	name := config.ContainerName
 	rootfs := filepath.Join(lxc_dir, name, "rootfs")
 
-	if config.PackerForce || config.CleanupFirst {
-		s.destroy(config.ContainerName, ui)
+	if config.PackerForce {
+		s.destroy(name, ui)
 	}
 
-	commands := make([][]string, 3)
-	if config.CloneSource == "" {
-		commands[0] = append(config.EnvVars, []string{"lxc-create", "-n", name, "-t", config.Name, "--"}...)
-		commands[0] = append(commands[0], config.Parameters...)
+	if config.LxcTemplate.Name != "" {
+		commands := make([][]string, 2)
+		commands[0] = append(config.LxcTemplate.EnvVars, []string{"lxc-create", "-n", name, "-t", config.LxcTemplate.Name, "--"}...)
+		commands[0] = append(commands[0], config.LxcTemplate.Parameters...)
+		// prevent tmp from being cleaned on boot, we put provisioning scripts there
+		// TODO: wait for init to finish before moving on to provisioning instead of this
+		commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
+
+		ui.Say("Creating container from template...")
+		for _, command := range commands {
+			log.Printf("Executing sudo command: %#v", command)
+			if err := s.SudoCommand(command...); err != nil {
+				errorHandler(fmt.Errorf("Command (%s) failed with error: %s", strings.Join(command, " "), err))
+				return multistep.ActionHalt
+			}
+		}
 	} else {
-		commands[0] = []string{"lxc-clone", "--orig", config.CloneSource, "--new", name}
-	}
-	// prevent tmp from being cleaned on boot, we put provisioning scripts there
-	// todo: wait for init to finish before moving on to provisioning instead of this
-	commands[1] = []string{"touch", filepath.Join(rootfs, "tmp", ".tmpfs")}
-	commands[2] = []string{"lxc-start", "-d", "--name", name}
-
-	ui.Say("Creating container...")
-	for _, command := range commands {
-		log.Printf("Executing sudo command: %#v", command)
-		err := s.SudoCommand(command...)
+		containerPath := filepath.Join(lxc_dir, name)
+		containerConfig, err := NewLxcConfig(config.RootFs.ConfigFile)
+		if err != nil {
+			errorHandler(fmt.Errorf("Could not read lxc config (%s): %s", config.RootFs.ConfigFile, err))
+			return multistep.ActionHalt
+		}
+		containerConfig.SetRootFs(rootfs)
+		tmpDir, err := ioutil.TempDir("", "lxcconfig")
 		if err != nil {
-			err := fmt.Errorf("Error creating container: %s", err)
-			state.Put("error", err)
-			ui.Error(err.Error())
+			errorHandler(fmt.Errorf("Could not create temp directory (%s): %s", tmpDir, err))
 			return multistep.ActionHalt
 		}
+		err = containerConfig.Write(filepath.Join(tmpDir, "lxc.config"))
```

Comment:
It'd be really nice to write this to the correct directory in the first place. I think go has facilities to make paths and you know where it _will_ be already so you might be able to do this without a temporary directory.